source/mir/random/discrete.d

```diff
@@ -68,7 +65,6 @@ struct Discrete(T)
     this(const(T)[] cdPoints)
     {
         _cdPoints = cdPoints;
-        r = typeof(r)(_cdPoints);
```

Comment:
Maybe something to do in a separate PR, but I think that `cdPoints` can be generalized to a forward range, and it should probably be `save`d to make sure it belongs to the `Discrete` distribution (cf. my point in the earlier issue that `const(T)[]` gives insufficient guarantees that your distribution will not change over time).

The use of `assumeSorted` in `opCall` should at least provide better debug checks that the fundamental property of sortedness isn't being violated, though.